File: backend/src/main/java/com/thanhtam/backend/dto/UserUpdate.java
package com.thanhtam.backend.dto;

import com.thanhtam.backend.entity.Profile;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class UserUpdate {
    private String username;
    private String email;
    private String password;
    private Profile profile;
}

File: source code/front-end-user/src/app/models/user-update.ts
import {UserRole} from './user-role';
import {UserProfile} from './user-profile';

export class UserUpdate {
  username: string;
  email: string;
  password: string;
  profile: UserProfile;


  constructor(username: string, email: string, password: string, profile: UserProfile) {
    this.username = username;
    this.email = email;
    this.password = password;
    this.profile = profile;
  }
}
